src/libcollections/btree/map.rs

```diff
@@ -276,6 +344,14 @@ pub struct OccupiedEntry<'a, K: 'a, V: 'a> {
 
 impl<K: Ord, V> BTreeMap<K, V> {
     /// Makes a new empty BTreeMap with a reasonable choice for B.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::collections::BTreeMap;
+    ///
+    /// let mut map: BTreeMap<&str, isize> = BTreeMap::new();
```

Comment:
This is me thinking out loud, but usually, the type of the map is inferred from usage. So we have two options:

1. show it like this, with an unneeded annotation
2. add a call to insert, which is sort of extraneous but is more realistic

I'm torn. maybe the second, but with a brief comment about it?